pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -287,3 +288,35 @@ func (s *NSXServiceAccountService) GetNSXServiceAccountNameByUID(uid string) (na
 func (s *NSXServiceAccountService) getClusterName(namespace, name string) string {
 	return fmt.Sprintf("%s-%s-%s", s.NSXConfig.CoeConfig.Cluster, namespace, name)
 }
+
+func GenerateNSXServiceAccountConditions(existingConditions []metav1.Condition, generation int64, realizedStatus metav1.ConditionStatus, reason string) []metav1.Condition {
+	var conditions []metav1.Condition
+	for i := range existingConditions {
+		switch existingConditions[i].Type {
+		case v1alpha1.ConditionTypeRealized:
+		default:
+			conditions = append(conditions, *existingConditions[i].DeepCopy())
+		}
+	}
+	conditions = append(conditions, metav1.Condition{
+		Type:               v1alpha1.ConditionTypeRealized,
+		Status:             realizedStatus,
+		ObservedGeneration: generation,
+		LastTransitionTime: metav1.Now(),
```

Comment:
If the condition status doesn't change, should it copy the LastTransitionTime from the existing ConditionTypeRealized condition?

Reply:
No, this time means the timestamp NSXServiceAccount controller process this NSXServiceAccount, and this func will be called only when controller finished the process.

Reply:
According to my understanding of the doc comment in https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition ,  LastTransitionTime should not be updated if the status doesn't change.

> // lastTransitionTime is the last time the condition transitioned from one status to another.
> // This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.

Reply:
Fixed. Will use existing LastTransitionTime if status is not changed.